### PR TITLE
2311 embed keyboard shortcuts

### LINKED ISSUE #2311: Embed keyboard shortcuts within the app
## Scope & Context

We want to embed keyboard shortcuts within the app (like Gmail, Linear, etc., do), something similar to:

keyboard shortcuts example

## Current behavior

Currently, we only have shortcuts related to the command bar. 

## Expected behavior

There can be 2 approaches:

1. User hits the ? key anywhere within the app, which brings up a list/panel of available shortcuts
2. Have a '?' somewhere at the bottom of the screen, which opens up a menu with a link to the 'shortcuts'. Something similar to:
 
image

Clicking on shortcuts should open up a panel with all available shortcuts.  

A possible enhancement in both cases is to have the ability to search for shortcuts as well.

cc @FelixMalfait @Bonapara

### PR DESCRIPTION
**Pull request by Kanav-Arora**
Closes #2311 

The shortcuts are not hard coded and can be omitted with change in list.

Please find below a walkthrough of the same

Menu can be opened using `Shift + ?` and can be closed by `Shift + ?` or cross button click.

https://github.com/twentyhq/twenty/assets/62852363/b1ea7191-c6c1-436d-a050-9587489613a0

### CODE REVIEW

File: front/src/modules/keyboard-shortcut-menu/components/KeyboardShortcutMenuStyles.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import styled from '@emotion/styled';
+
+import { useIsMobile } from '@/ui/utilities/responsive/hooks/useIsMobile';
+
+export const StyledDialog = styled.div`
```
Comment (charlesBochet) [inline front/src/modules/keyboard-shortcut-menu/components/KeyboardShortcutMenuStyles.tsx:5]: we should leverage our existing Dialog component here. You can take a look at the DialogManager

File: front/src/modules/keyboard-shortcut-menu/components/KeyboardShortcutMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import { KeyboardMenuDialog } from './KeyboardShortcutMenuDialog';
+import { KeyboardMenuGroup } from './KeyboardShortcutMenuGroup';
+import { KeyboardMenuItem } from './KeyboardShortcutMenuItem';
+
+export const KeyboardShortcutMenu = () => {
+  const { toggleKeyboardShortcutMenu } = useKeyboardShortcutMenu();
+  const isKeyboardShortcutMenuOpened = useRecoilValue(
+    isKeyboardShortcutMenuOpenedState,
+  );
+  useScopedHotkeys(
```
Comment (charlesBochet) [inline front/src/modules/keyboard-shortcut-menu/components/KeyboardShortcutMenu.tsx:22]: escape should also close the menu

**Review by charlesBochet — COMMENTED** [review]
The code quality is very good.
I've left comment.
Also could you take another look at the design, I see many differences in the size of elements, font-color...

**Comment by charlesBochet** [comment]
And thank you @Kanav-Arora, much appreciated :)

**Comment by Kanav-Arora** [comment]
Hi @charlesBochet 
Thanks!

I have done most of the requested changes
I'm unable to override Dialog Manager styles with styled components.

**Review by charlesBochet — APPROVED** [review]
Looks great! Thank you, it's a nice work :)

**Pull request merged**